src/error.rs

```diff
@@ -8,14 +8,44 @@ pub enum ASEError {
     /// An error was encountered while parsing the ASE.
     ///
     /// This means that the input data did not conform to the ASE specification.
-    Invalid,
+    Invalid(ConformationError),
+    /// An error occured due to slice conversion issues.
+    SliceError,
```

Comment:
I'm not sure about this error, it feels a bit too generic, and does not really give the user any clue, about what component failed to get parsed. Will have to think about it a bit more.